src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -189,7 +185,7 @@ public Visitable visit(MethodDeclaration method, Void p) {
     // TODO: test this with annotations
     String methodName = this.classFQName + "#" + removeMethodReturnType(methodDeclAsString);
     // this method belongs to an anonymous class inside the target method
-    if (insideTargetMethod) {
+    if (insideTargetMethod && method.getParentNode().isPresent()) {
```

Comment:
I think the second condition check was unneeded. `method` will definitely have a parent node, since it is inside another method.

Reply:
Yes, this should have been removed in later commits but forgot. Nice catch!